Sephora-M/assignment2/Main.java

```diff
@@ -0,0 +1,28 @@
+package assignment2;
+
+public class Main {
+
+	public static void main(String[] args) {
+		BinaryTreeNode<String> a = new BinaryTreeNode<String>("a", null);
+		BinaryTree<String> A = new BinaryTree<String>(a);
+		
+		String[] keys = {"b","c","d","e","f","g","h","i","j","k","l","m", "n"};
+		
+		for (String  key : keys) {
+			A.add(key);
```

Comment:
Due to this method being randomised the output of the "common ancestor of (x, y)" in lines 17-24 will also be changing, is that correct?

Reply:
Yes that's right. That is why I handcrafted a tree in the JUnit test. 
But now with the method that takes a in order description I can also do this more elegantly. I've added an example in the main and in the test.